[package] freetype/2.13.3 : Meson does not respect Conan host profile arch in cross-compilation

### Description

 When attempting to cross-compile `freetype/2.13.3` (this version uses Meson as the build system), the architecture specified in the host profile, for example `arch=x86` is not respected.
 This results in the build being configured for the build machine's architecture instead of the host architecture, failing the linkage:
```
[46/46] Linking target libfreetype.so.6.20.2
FAILED: libfreetype.so.6.20.2 
gcc  -o libfreetype.so.6.20.2 libfreetype.so.6.20.2.p/src_base_ftbase.c.o libfreetype.so.6.20.2.p/src_base_ftinit.c.o ...
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(inflate.c.o)' is incompatible with i386:x86-64 output
```

 Looking at the log detail (see Logs section), and the generated meson config it seems that did not add the `-m32` flag when arch=`x86`:
```
cat /home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release/conan/conan_meson_cross.ini
[properties]
needs_exe_wrapper = true

[constants]
preprocessor_definitions = []

[project options]
wrap_mode = 'nofallback'
bindir = 'bin'
sbindir = 'bin'
libexecdir = 'bin'
includedir = 'include'
libdir = 'lib'
brotli = 'enabled'
bzip2 = 'enabled'
harfbuzz = 'disabled'
png = 'enabled'
tests = 'disabled'
zlib = 'system'


[binaries]
c = 'gcc'
cpp = 'g++'
ld = 'gcc'
ar = 'ar'
strip = 'strip'
as = 'as'
pkgconfig = '/home/amachado/.conan2/p/b/pkgcodf9bc1fd0fd07/p/bin/pkgconf'
pkg-config = '/home/amachado/.conan2/p/b/pkgcodf9bc1fd0fd07/p/bin/pkgconf'

[built-in options]
buildtype = 'release'
default_library = 'shared'
b_ndebug = 'true'
backend = 'ninja'
pkg_config_path = '/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release/conan'
# C/C++ arguments
c_args = [] + preprocessor_definitions
c_link_args = []
cpp_args = [] + preprocessor_definitions
cpp_link_args = []

[build_machine]
system = 'linux'
cpu_family = 'x86_64'
cpu = 'x86_64'
endian = 'little'
[host_machine]
system = 'linux'
cpu_family = 'x86'
cpu = 'x86'
endian = 'little'
```

**Workaround**:
If I force the `-m32` flag in the profile :host the compilation is successful:
```
[buildenv]
CFLAGS= -m32
CXXFLAGS= -m32
LDFLAGS= -m32
```

```
cat /home/amachado/.conan2/p/b/freet2d65bd924b0fd/b/build-release/conan/conan_meson_cross.ini
[properties]
needs_exe_wrapper = true

[constants]
preprocessor_definitions = []

[project options]
wrap_mode = 'nofallback'
bindir = 'bin'
sbindir = 'bin'
libexecdir = 'bin'
includedir = 'include'
libdir = 'lib'
brotli = 'enabled'
bzip2 = 'enabled'
harfbuzz = 'disabled'
png = 'enabled'
tests = 'disabled'
zlib = 'system'


[binaries]
c = 'gcc'
cpp = 'g++'
ld = 'gcc'
ar = 'ar'
strip = 'strip'
as = 'as'
pkgconfig = '/home/amachado/.conan2/p/b/pkgcodf9bc1fd0fd07/p/bin/pkgconf'
pkg-config = '/home/amachado/.conan2/p/b/pkgcodf9bc1fd0fd07/p/bin/pkgconf'

[built-in options]
buildtype = 'release'
default_library = 'shared'
b_ndebug = 'true'
backend = 'ninja'
pkg_config_path = '/home/amachado/.conan2/p/b/freet2d65bd924b0fd/b/build-release/conan'
# C/C++ arguments
c_args = ['-m32'] + preprocessor_definitions
c_link_args = ['-m32']
cpp_args = ['-m32'] + preprocessor_definitions
cpp_link_args = ['-m32']

[build_machine]
system = 'linux'
cpu_family = 'x86_64'
cpu = 'x86_64'
endian = 'little'
[host_machine]
system = 'linux'
cpu_family = 'x86'
cpu = 'x86'
endian = 'little'
```

But that is redundant with `arch=x86` defined in the Conan host profile that Conan uses in the recipes to inject defining architecture flag: -m32 in toolchains

### Package and Environment Details

* Package Name/Version: **freetype/2.13.3**
* Operating System+version: **Linux Ubuntu 22.04.5 LTS**
* Compiler+version: **GCC 11.4.0**

* Conan version: **conan 2.11.0**
* Python version: **Python 3.10.12**


### Conan profile

build on Linux-86_64 (64-bit) to host: Linux x86 (32-bit)

`--profile:build=default`
```
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
```

`--profile:host=linux-x86`
```
[settings]
os=Linux
arch=x86
build_type=Release
compiler=gcc
compiler.cppstd=11
compiler.libcxx=libstdc++11
compiler.version=6.3

[conf]
tools.cmake.cmaketoolchain:user_toolchain=["{{profile_dir}}/../cmake/toolchains/linux-x86.toolchain.cmake"]

[buildenv]
AS=as
LD=gcc
CC=gcc
CXX=g++
AR=ar
NM=nm
STRIP=strip
OBJCOPY=objcopy
OBJDUMP=objdump
RANLIB=ranlib
```

`linux-x86.toolchain.cmake`
```
set(CMAKE_SYSTEM_NAME Linux)
set(CMAKE_SYSTEM_PROCESSOR x86)

set(CMAKE_C_COMPILER gcc)
set(CMAKE_CXX_COMPILER g++)
set(CMAKE_ASM_COMPILER as)
set(CMAKE_AR ar)
set(CMAKE_NM nm)
set(CMAKE_RANLIB ranlib)
set(CMAKE_STRIP strip)
set(CMAKE_OBJCOPY objcopy)
set(CMAKE_OBJDUMP objdump)
```

### Steps to reproduce

In the recipe folder of the cloned repo `conan-center-index/recipes/freetype`:
```
conan create meson --version=2.13.3 -pr:b=default -pr:h=linux-x86 -o "&:shared=True" --build=missing
```

### Logs

<details><summary>Click to expand log</summary>

```
conan create meson --version=2.13.3 -pr:b=default -pr:h=/home/amachado/code/winrest/conan_profiles/linux-x86 -o "&
:shared=True" --build=missing

======== Exporting recipe to the cache ========
freetype/2.13.3: Exporting package recipe: /home/amachado/code/conan-center-index/recipes/freetype/meson/conanfile.py
freetype/2.13.3: exports: File 'conandata.yml' found. Exporting it...
freetype/2.13.3: Calling export_sources()
freetype/2.13.3: Copied 1 '.py' file: conanfile.py
freetype/2.13.3: Copied 1 '.yml' file: conandata.yml
freetype/2.13.3: Copied 1 '.patch' file: 2.13.3-0002-meson-Fix-static-windows.patch
freetype/2.13.3: Exported to cache folder: /home/amachado/.conan2/p/freet4b24eade8195d/e
freetype/2.13.3: Exported: freetype/2.13.3#a5f673aa074a44a0eedb23c901727d11 (2025-01-09 11:25:04 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=x86
build_type=Release
compiler=gcc
compiler.cppstd=11
compiler.libcxx=libstdc++11
compiler.version=6.3
os=Linux
[options]
&:shared=True
[conf]
tools.cmake.cmaketoolchain:user_toolchain=['/home/amachado/code/winrest/conan_profiles/../cmake/toolchains/linux-x86.toolchain.cmake']
[buildenv]
AS=as
LD=gcc
CC=gcc
CXX=g++
AR=ar
NM=nm
STRIP=strip
OBJCOPY=objcopy
OBJDUMP=objdump
RANLIB=ranlib

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux


======== Computing dependency graph ========
libpng/1.6.44: Not found in local cache, looking in remotes...
libpng/1.6.44: Checking remote: conancenter
libpng/1.6.44: Downloaded recipe revision 9e1aa08fb46946c7c91e4ae03bd49811
bzip2/1.0.8: Not found in local cache, looking in remotes...
bzip2/1.0.8: Checking remote: conancenter
bzip2/1.0.8: Downloaded recipe revision d00dac990f08d991998d624be81a9526
brotli/1.1.0: Not found in local cache, looking in remotes...
brotli/1.1.0: Checking remote: conancenter
brotli/1.1.0: Downloaded recipe revision d56d7bb9ca722942aba17369cb5c0519
Graph root
    cli
Requirements
    brotli/1.1.0#d56d7bb9ca722942aba17369cb5c0519 - Downloaded (conancenter)
    bzip2/1.0.8#d00dac990f08d991998d624be81a9526 - Downloaded (conancenter)
    freetype/2.13.3#a5f673aa074a44a0eedb23c901727d11 - Cache
    libpng/1.6.44#9e1aa08fb46946c7c91e4ae03bd49811 - Downloaded (conancenter)
    zlib/1.3.1#b8bc2603263cf7eccbd6e17e66b0ed76 - Cache
Build requirements
    meson/1.2.2#21b73818ba96d9eea465b310b5bbc993 - Cache
    meson/1.3.2#26ce8a76a36cc275cdfee1d757bc6561 - Cache
    ninja/1.12.1#fd583651bf0c6a901943495d49878803 - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache
Resolved version ranges
    libpng/[>=1.6 <2]: libpng/1.6.44
    ninja/[>=1.10.2 <2]: ninja/1.12.1
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
Requirements
    brotli/1.1.0#d56d7bb9ca722942aba17369cb5c0519:2ead8b9ed9193280857d4545d3c0e40001079d11 - Build
    bzip2/1.0.8#d00dac990f08d991998d624be81a9526:918476d4a378c4df190725a424695b0b2631b58c - Build
    freetype/2.13.3#a5f673aa074a44a0eedb23c901727d11:5382c7fd93cc143dabc99020ee69af8fd104786c - Build
    libpng/1.6.44#9e1aa08fb46946c7c91e4ae03bd49811:ebd7fe5cba2e2da22231411c89fbc2400b15b5d9 - Build
    zlib/1.3.1#b8bc2603263cf7eccbd6e17e66b0ed76:41a9ad057d6605aab5f50650c1aa915724c4fca4#de3800450ce3b23f870359a0b1a61a0b - Cache
Build requirements
    meson/1.3.2#26ce8a76a36cc275cdfee1d757bc6561:da39a3ee5e6b4b0d3255bfef95601890afd80709#3ba677cf44c95996f4f326c668f92f00 - Cache
    ninja/1.12.1#fd583651bf0c6a901943495d49878803:3593751651824fb813502c69c971267624ced41a#20d1a5badf48926f22c4c55d097392ca - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605:c0b621fd4b3199fe05075171573398833dba85f4#3c6350f5ba8a9c43a2a29adcf47bac45 - Cache
Skipped binaries
    meson/1.2.2

======== Installing packages ========
brotli/1.1.0: Sources downloaded from 'conancenter'
brotli/1.1.0: Calling source() in /home/amachado/.conan2/p/brotl01cfbaf421d56/s/src
brotli/1.1.0: Unzipping v1.1.0.tar.gz to .

-------- Installing package brotli/1.1.0 (1 of 8) --------
brotli/1.1.0: Building from source
brotli/1.1.0: Package brotli/1.1.0:2ead8b9ed9193280857d4545d3c0e40001079d11
brotli/1.1.0: Copying sources to build folder
brotli/1.1.0: Building your package in /home/amachado/.conan2/p/b/brotl7a1ee101c276c/b
brotli/1.1.0: Calling generate()
brotli/1.1.0: Generators folder: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/build/Release/generators
brotli/1.1.0: CMakeToolchain generated: conan_toolchain.cmake
brotli/1.1.0: CMakeToolchain generated: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/build/Release/generators/CMakePresets.json
brotli/1.1.0: CMakeToolchain generated: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/src/CMakeUserPresets.json
brotli/1.1.0: Generating aggregated env files
brotli/1.1.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
brotli/1.1.0: Calling build()
brotli/1.1.0: Apply patch (conan): add target properties, disable command build
brotli/1.1.0: Running CMake.configure()
brotli/1.1.0: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/src"
-- Using Conan toolchain: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Including user_toolchain: /home/amachado/code/winrest/conan_profiles/../cmake/toolchains/linux-x86.toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Defining architecture flag: -m32
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Build type is 'Release'
-- Performing Test BROTLI_EMSCRIPTEN
-- Performing Test BROTLI_EMSCRIPTEN - Failed
-- Compiler is not EMSCRIPTEN
-- Looking for log2
-- Looking for log2 - not found
-- Looking for log2
-- Looking for log2 - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/build/Release

brotli/1.1.0: Running CMake.build()
brotli/1.1.0: RUN: cmake --build "/home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/build/Release" -- -j8
[  8%] Building C object CMakeFiles/brotlicommon.dir/c/common/dictionary.c.o
[  8%] Building C object CMakeFiles/brotlicommon.dir/c/common/context.c.o
[  8%] Building C object CMakeFiles/brotlicommon.dir/c/common/constants.c.o
[ 11%] Building C object CMakeFiles/brotlicommon.dir/c/common/platform.c.o
[ 14%] Building C object CMakeFiles/brotlicommon.dir/c/common/shared_dictionary.c.o
[ 17%] Building C object CMakeFiles/brotlicommon.dir/c/common/transform.c.o
[ 20%] Linking C static library libbrotlicommon.a
[ 20%] Built target brotlicommon
[ 26%] Building C object CMakeFiles/brotlidec.dir/c/dec/huffman.c.o
[ 26%] Building C object CMakeFiles/brotlidec.dir/c/dec/decode.c.o
[ 29%] Building C object CMakeFiles/brotlidec.dir/c/dec/bit_reader.c.o
[ 32%] Building C object CMakeFiles/brotlienc.dir/c/enc/backward_references.c.o
[ 35%] Building C object CMakeFiles/brotlidec.dir/c/dec/state.c.o
[ 38%] Building C object CMakeFiles/brotlienc.dir/c/enc/backward_references_hq.c.o
[ 41%] Building C object CMakeFiles/brotlienc.dir/c/enc/bit_cost.c.o
[ 44%] Building C object CMakeFiles/brotlienc.dir/c/enc/block_splitter.c.o
[ 47%] Building C object CMakeFiles/brotlienc.dir/c/enc/brotli_bit_stream.c.o
[ 50%] Building C object CMakeFiles/brotlienc.dir/c/enc/cluster.c.o
[ 52%] Building C object CMakeFiles/brotlienc.dir/c/enc/command.c.o
[ 55%] Building C object CMakeFiles/brotlienc.dir/c/enc/compound_dictionary.c.o
[ 58%] Building C object CMakeFiles/brotlienc.dir/c/enc/compress_fragment.c.o
[ 61%] Building C object CMakeFiles/brotlienc.dir/c/enc/compress_fragment_two_pass.c.o
[ 64%] Building C object CMakeFiles/brotlienc.dir/c/enc/dictionary_hash.c.o
[ 67%] Building C object CMakeFiles/brotlienc.dir/c/enc/encode.c.o
[ 70%] Building C object CMakeFiles/brotlienc.dir/c/enc/encoder_dict.c.o
[ 73%] Building C object CMakeFiles/brotlienc.dir/c/enc/entropy_encode.c.o
[ 76%] Building C object CMakeFiles/brotlienc.dir/c/enc/fast_log.c.o
[ 79%] Building C object CMakeFiles/brotlienc.dir/c/enc/histogram.c.o
[ 82%] Building C object CMakeFiles/brotlienc.dir/c/enc/literal_cost.c.o
[ 85%] Building C object CMakeFiles/brotlienc.dir/c/enc/memory.c.o
[ 88%] Linking C static library libbrotlidec.a
[ 91%] Building C object CMakeFiles/brotlienc.dir/c/enc/metablock.c.o
[ 91%] Built target brotlidec
[ 94%] Building C object CMakeFiles/brotlienc.dir/c/enc/static_dict.c.o
[ 97%] Building C object CMakeFiles/brotlienc.dir/c/enc/utf8_util.c.o
[100%] Linking C static library libbrotlienc.a
[100%] Built target brotlienc

brotli/1.1.0: Package '2ead8b9ed9193280857d4545d3c0e40001079d11' built
brotli/1.1.0: Build folder /home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/build/Release
brotli/1.1.0: Generating the package
brotli/1.1.0: Packaging in folder /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p
brotli/1.1.0: Calling package()
brotli/1.1.0: Running CMake.install()
brotli/1.1.0: RUN: cmake --install "/home/amachado/.conan2/p/b/brotl7a1ee101c276c/b/build/Release" --prefix "/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p"
-- Install configuration: "Release"
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlienc.a
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlidec.a
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/include/brotli
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/include/brotli/shared_dictionary.h
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/include/brotli/port.h
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/include/brotli/types.h
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/include/brotli/encode.h
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/include/brotli/decode.h
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/pkgconfig/libbrotlicommon.pc
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/pkgconfig/libbrotlidec.pc
-- Installing: /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/pkgconfig/libbrotlienc.pc

brotli/1.1.0: package(): Packaged 1 file: LICENSE
brotli/1.1.0: package(): Packaged 3 '.a' files: libbrotlienc.a, libbrotlicommon.a, libbrotlidec.a
brotli/1.1.0: package(): Packaged 5 '.h' files
brotli/1.1.0: Created package revision e2084a1f73e07859ff721b16e888e26d
brotli/1.1.0: Package '2ead8b9ed9193280857d4545d3c0e40001079d11' created
brotli/1.1.0: Full package reference: brotli/1.1.0#d56d7bb9ca722942aba17369cb5c0519:2ead8b9ed9193280857d4545d3c0e40001079d11#e2084a1f73e07859ff721b16e888e26d
brotli/1.1.0: Package folder /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p
bzip2/1.0.8: Sources downloaded from 'conancenter'
bzip2/1.0.8: Calling source() in /home/amachado/.conan2/p/bzip22a75d000ecba1/s/src
bzip2/1.0.8: Unzipping bzip2-1.0.8.tar.gz to .

-------- Installing package bzip2/1.0.8 (2 of 8) --------
bzip2/1.0.8: Building from source
bzip2/1.0.8: Package bzip2/1.0.8:918476d4a378c4df190725a424695b0b2631b58c
bzip2/1.0.8: Copying sources to build folder
bzip2/1.0.8: Building your package in /home/amachado/.conan2/p/b/bzip27630913063559/b
bzip2/1.0.8: Calling generate()
bzip2/1.0.8: Generators folder: /home/amachado/.conan2/p/b/bzip27630913063559/b/build/Release/generators
bzip2/1.0.8: CMakeToolchain generated: conan_toolchain.cmake
bzip2/1.0.8: CMakeToolchain generated: /home/amachado/.conan2/p/b/bzip27630913063559/b/build/Release/generators/CMakePresets.json
bzip2/1.0.8: Generating aggregated env files
bzip2/1.0.8: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
bzip2/1.0.8: Calling build()
bzip2/1.0.8: apply_conandata_patches(): No patches defined in conandata
bzip2/1.0.8: Running CMake.configure()
bzip2/1.0.8: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/amachado/.conan2/p/b/bzip27630913063559/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/amachado/.conan2/p/b/bzip27630913063559/b/src/.."
-- Using Conan toolchain: /home/amachado/.conan2/p/b/bzip27630913063559/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Including user_toolchain: /home/amachado/code/winrest/conan_profiles/../cmake/toolchains/linux-x86.toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Defining architecture flag: -m32
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/amachado/.conan2/p/b/bzip27630913063559/b/build/Release

bzip2/1.0.8: Running CMake.build()
bzip2/1.0.8: RUN: cmake --build "/home/amachado/.conan2/p/b/bzip27630913063559/b/build/Release" -- -j8
[ 10%] Building C object CMakeFiles/bz2.dir/src/decompress.c.o
[ 20%] Building C object CMakeFiles/bz2.dir/src/blocksort.c.o
[ 40%] Building C object CMakeFiles/bz2.dir/src/compress.c.o
[ 40%] Building C object CMakeFiles/bz2.dir/src/bzlib.c.o
[ 50%] Building C object CMakeFiles/bz2.dir/src/crctable.c.o
[ 60%] Building C object CMakeFiles/bz2.dir/src/huffman.c.o
[ 70%] Building C object CMakeFiles/bz2.dir/src/randtable.c.o
[ 80%] Linking C static library libbz2.a
[ 80%] Built target bz2
[ 90%] Building C object CMakeFiles/bzip2.dir/src/bzip2.c.o
/home/amachado/.conan2/p/b/bzip27630913063559/b/src/bzip2.c: In function ‘applySavedFileAttrToOutputFile’:
/home/amachado/.conan2/p/b/bzip27630913063559/b/src/bzip2.c:1073:11: warning: ignoring return value of ‘fchown’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1073 |    (void) fchown ( fd, fileMetaInfo.st_uid, fileMetaInfo.st_gid );
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[100%] Linking C executable bzip2
[100%] Built target bzip2

bzip2/1.0.8: Package '918476d4a378c4df190725a424695b0b2631b58c' built
bzip2/1.0.8: Build folder /home/amachado/.conan2/p/b/bzip27630913063559/b/build/Release
bzip2/1.0.8: Generating the package
bzip2/1.0.8: Packaging in folder /home/amachado/.conan2/p/b/bzip27630913063559/p
bzip2/1.0.8: Calling package()
bzip2/1.0.8: Running CMake.install()
bzip2/1.0.8: RUN: cmake --install "/home/amachado/.conan2/p/b/bzip27630913063559/b/build/Release" --prefix "/home/amachado/.conan2/p/b/bzip27630913063559/p"
-- Install configuration: "Release"
-- Installing: /home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a
-- Installing: /home/amachado/.conan2/p/b/bzip27630913063559/p/include/bzlib.h
-- Installing: /home/amachado/.conan2/p/b/bzip27630913063559/p/bin/bzip2

bzip2/1.0.8: package(): Packaged 2 files: LICENSE, bzip2
bzip2/1.0.8: package(): Packaged 1 '.a' file: libbz2.a
bzip2/1.0.8: package(): Packaged 1 '.cmake' file: conan-official-bzip2-variables.cmake
bzip2/1.0.8: package(): Packaged 1 '.h' file: bzlib.h
bzip2/1.0.8: Created package revision 20c2abbade1ebd806d322edf3e8a0499
bzip2/1.0.8: Package '918476d4a378c4df190725a424695b0b2631b58c' created
bzip2/1.0.8: Full package reference: bzip2/1.0.8#d00dac990f08d991998d624be81a9526:918476d4a378c4df190725a424695b0b2631b58c#20c2abbade1ebd806d322edf3e8a0499
bzip2/1.0.8: Package folder /home/amachado/.conan2/p/b/bzip27630913063559/p
ninja/1.12.1: Already installed! (3 of 8)
zlib/1.3.1: Already installed! (4 of 8)
pkgconf/2.1.0: Already installed! (5 of 8)
pkgconf/2.1.0: WARN: The use of 'unix_path_legacy_compat' is deprecated in Conan 2.0 and does not perform path conversions. This is retained for compatibility with Conan 1.x and will be removed in a future version.
libpng/1.6.44: Sources downloaded from 'conancenter'
libpng/1.6.44: Calling source() in /home/amachado/.conan2/p/libpn96b6230f1a784/s/src
libpng/1.6.44: Unzipping libpng-1.6.44.tar.xz to .

-------- Installing package libpng/1.6.44 (6 of 8) --------
libpng/1.6.44: Building from source
libpng/1.6.44: Package libpng/1.6.44:ebd7fe5cba2e2da22231411c89fbc2400b15b5d9
libpng/1.6.44: Copying sources to build folder
libpng/1.6.44: Building your package in /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b
libpng/1.6.44: Calling generate()
libpng/1.6.44: Generators folder: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/build/Release/generators
libpng/1.6.44: CMakeToolchain generated: conan_toolchain.cmake
libpng/1.6.44: CMakeToolchain generated: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/build/Release/generators/CMakePresets.json
libpng/1.6.44: CMakeToolchain generated: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/src/CMakeUserPresets.json
libpng/1.6.44: Generating aggregated env files
libpng/1.6.44: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
libpng/1.6.44: Calling build()
libpng/1.6.44: apply_conandata_patches(): No patches defined in conandata
libpng/1.6.44: Running CMake.configure()
libpng/1.6.44: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p" -DPNG_TESTS="OFF" -DPNG_SHARED="OFF" -DPNG_STATIC="ON" -DPNG_DEBUG="OFF" -DPNG_PREFIX="" -DCMAKE_MACOSX_BUNDLE="OFF" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/src"
-- Using Conan toolchain: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Including user_toolchain: /home/amachado/code/winrest/conan_profiles/../cmake/toolchains/linux-x86.toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Defining architecture flag: -m32
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is GNU 11.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/as
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Building for target architecture: x86
-- Conan: Target declared 'ZLIB::ZLIB'
-- Performing Test HAVE_LD_VERSION_SCRIPT
-- Performing Test HAVE_LD_VERSION_SCRIPT - Success
-- Found AWK program: /usr/bin/gawk
-- Configuring done
-- Generating done
-- Build files have been written to: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/build/Release

libpng/1.6.44: Running CMake.build()
libpng/1.6.44: RUN: cmake --build "/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/build/Release" -- -j8
[  3%] Generating scripts/symbols.out
[  3%] Generating pnglibconf.c
[  4%] Generating scripts/pnglibconf.c
[  6%] Generating pngprefix.h
[  6%] Built target pngprefix_h
[  6%] Built target pnglibconf_c
[  6%] Built target png_scripts_pnglibconf_c
[  6%] Built target png_scripts_symbols_out
[  7%] Generating pnglibconf.out
[  9%] Generating scripts/symbols.chk
[ 11%] Built target png_scripts_symbols_chk
[ 12%] Built target pnglibconf_out
[ 15%] Generating pnglibconf.h
[ 15%] Generating scripts/prefix.out
[ 19%] Built target pnglibconf_h
[ 22%] Built target png_scripts_prefix_out
[ 23%] Generating scripts/vers.out
[ 25%] Generating scripts/intprefix.out
[ 26%] Generating scripts/sym.out
[ 31%] Built target png_scripts_vers_out
[ 36%] Built target png_scripts_sym_out
[ 39%] Generating libpng.vers
[ 39%] Generating libpng.sym
[ 44%] Built target png_scripts_intprefix_out
[ 44%] Built target png_gensym
[ 44%] Built target png_genvers
[ 65%] Built target png_genfiles
[ 71%] Building C object CMakeFiles/png_static.dir/pngget.c.o
[ 71%] Building C object CMakeFiles/png_static.dir/pngmem.c.o
[ 73%] Building C object CMakeFiles/png_static.dir/pngpread.c.o
[ 74%] Building C object CMakeFiles/png_static.dir/pngrio.c.o
[ 77%] Building C object CMakeFiles/png_static.dir/pngrtran.c.o
[ 77%] Building C object CMakeFiles/png_static.dir/pngread.c.o
[ 71%] Building C object CMakeFiles/png_static.dir/pngerror.c.o
[ 71%] Building C object CMakeFiles/png_static.dir/png.c.o
[ 79%] Building C object CMakeFiles/png_static.dir/pngrutil.c.o
[ 80%] Building C object CMakeFiles/png_static.dir/pngset.c.o
[ 82%] Building C object CMakeFiles/png_static.dir/pngtrans.c.o
[ 84%] Building C object CMakeFiles/png_static.dir/pngwio.c.o
[ 85%] Building C object CMakeFiles/png_static.dir/pngwrite.c.o
[ 87%] Building C object CMakeFiles/png_static.dir/pngwtran.c.o
[ 88%] Building C object CMakeFiles/png_static.dir/pngwutil.c.o
[ 90%] Building C object CMakeFiles/png_static.dir/intel/intel_init.c.o
[ 92%] Building C object CMakeFiles/png_static.dir/intel/filter_sse2_intrinsics.c.o
[ 93%] Linking C static library libpng16.a
[100%] Built target png_static

libpng/1.6.44: Package 'ebd7fe5cba2e2da22231411c89fbc2400b15b5d9' built
libpng/1.6.44: Build folder /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/build/Release
libpng/1.6.44: Generating the package
libpng/1.6.44: Packaging in folder /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p
libpng/1.6.44: Calling package()
libpng/1.6.44: Running CMake.install()
libpng/1.6.44: RUN: cmake --install "/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/b/build/Release" --prefix "/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p"
-- Install configuration: "Release"
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng16.a
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/include/png.h
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/include/pngconf.h
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/include/pnglibconf.h
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/include/libpng16/png.h
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/include/libpng16/pngconf.h
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/include/libpng16/pnglibconf.h
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/bin/libpng-config
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/bin/libpng16-config
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/share/man/man3/libpng.3
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/share/man/man3/libpngpf.3
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/share/man/man5/png.5
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/pkgconfig/libpng.pc
-- Up-to-date: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/bin/libpng-config
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/pkgconfig/libpng16.pc
-- Up-to-date: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/bin/libpng16-config
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng/libpng16.cmake
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng/libpng16-release.cmake
-- Up-to-date: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng16.a
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/cmake/PNG/PNGTargets.cmake
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/cmake/PNG/PNGTargets-release.cmake
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/cmake/PNG/PNGConfig.cmake
-- Installing: /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/cmake/PNG/PNGConfigVersion.cmake

libpng/1.6.44: package(): Packaged 1 file: LICENSE
libpng/1.6.44: package(): Packaged 2 '.a' files: libpng16.a, libpng.a
libpng/1.6.44: package(): Packaged 6 '.h' files
libpng/1.6.44: Created package revision d8442a03abb8b96548c9fa49e5fd2ed9
libpng/1.6.44: Package 'ebd7fe5cba2e2da22231411c89fbc2400b15b5d9' created
libpng/1.6.44: Full package reference: libpng/1.6.44#9e1aa08fb46946c7c91e4ae03bd49811:ebd7fe5cba2e2da22231411c89fbc2400b15b5d9#d8442a03abb8b96548c9fa49e5fd2ed9
libpng/1.6.44: Package folder /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p
meson/1.3.2: Already installed! (7 of 8)
meson/1.3.2: Finalized folder /home/amachado/.conan2/p/meson592fcdbc462a9/f

-------- Installing package freetype/2.13.3 (8 of 8) --------
freetype/2.13.3: Building from source
freetype/2.13.3: Package freetype/2.13.3:5382c7fd93cc143dabc99020ee69af8fd104786c
freetype/2.13.3: Copying sources to build folder
freetype/2.13.3: Building your package in /home/amachado/.conan2/p/b/freet22b6a509fd61e/b
freetype/2.13.3: Calling generate()
freetype/2.13.3: Generators folder: /home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release/conan
freetype/2.13.3: MesonToolchain generated: conan_meson_cross.ini
freetype/2.13.3: Generating aggregated env files
freetype/2.13.3: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
freetype/2.13.3: Calling build()
freetype/2.13.3: Apply patch (portability): meson: define DLL_EXPORT for shared library only
freetype/2.13.3: Meson configure cmd: meson setup --cross-file "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release/conan/conan_meson_cross.ini" "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release" "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/src" --prefix=/
freetype/2.13.3: RUN: meson setup --cross-file "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release/conan/conan_meson_cross.ini" "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release" "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/src" --prefix=/
The Meson build system
Version: 1.3.2
Source dir: /home/amachado/.conan2/p/b/freet22b6a509fd61e/b/src
Build dir: /home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release
Build type: cross build
Project name: freetype2
Project version: 2.13.3
C compiler for the host machine: gcc (gcc 11.4.0 "gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C linker for the host machine: gcc ld.bfd 2.38
C compiler for the build machine: ccache cc (gcc 11.4.0 "cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C linker for the build machine: cc ld.bfd 2.38
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86
Host machine cpu: x86
Target machine cpu family: x86
Target machine cpu: x86
Program python3 found: YES (/home/amachado/code/winrest/virtualenvs/conan-v2/bin/python3)
Has header "unistd.h" : YES 
Has header "fcntl.h" : YES 
Has header "sys/mman.h" : YES 
Found pkg-config: YES (/home/amachado/.conan2/p/b/pkgcodf9bc1fd0fd07/p/bin/pkgconf) 2.1.0
Run-time dependency zlib found: YES 1.3.1
Run-time dependency bzip2 found: YES 1.0.8
Run-time dependency libpng found: YES 1.6.44
Dependency harfbuzz skipped: feature harfbuzz disabled
Run-time dependency libbrotlidec found: YES 1.1.0
Build targets in project: 5

freetype2 2.13.3

  Operating System
    OS         : linux

  Used Libraries
    Zlib       : system
    Bzip2      : yes
    Png        : yes
    Harfbuzz   : no
    Brotli     : yes

  User defined options
    Cross files: /home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release/conan/conan_meson_cross.ini
    prefix     : /

Found ninja-1.12.1 at /home/amachado/.conan2/p/b/ninjaab578b681fd73/p/bin/ninja

freetype/2.13.3: Meson build cmd: meson compile -C "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release" -j8
freetype/2.13.3: RUN: meson compile -C "/home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release" -j8
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/amachado/.conan2/p/b/ninjaab578b681fd73/p/bin/ninja -j 8
[46/46] Linking target libfreetype.so.6.20.2
FAILED: libfreetype.so.6.20.2 
gcc  -o libfreetype.so.6.20.2 libfreetype.so.6.20.2.p/src_base_ftbase.c.o libfreetype.so.6.20.2.p/src_base_ftinit.c.o libfreetype.so.6.20.2.p/src_truetype_truetype.c.o libfreetype.so.6.20.2.p/src_type1_type1.c.o libfreetype.so.6.20.2.p/src_cff_cff.c.o libfreetype.so.6.20.2.p/src_cid_type1cid.c.o libfreetype.so.6.20.2.p/src_pfr_pfr.c.o libfreetype.so.6.20.2.p/src_type42_type42.c.o libfreetype.so.6.20.2.p/src_winfonts_winfnt.c.o libfreetype.so.6.20.2.p/src_pcf_pcf.c.o libfreetype.so.6.20.2.p/src_bdf_bdf.c.o libfreetype.so.6.20.2.p/src_sfnt_sfnt.c.o libfreetype.so.6.20.2.p/src_autofit_autofit.c.o libfreetype.so.6.20.2.p/src_pshinter_pshinter.c.o libfreetype.so.6.20.2.p/src_smooth_smooth.c.o libfreetype.so.6.20.2.p/src_raster_raster.c.o libfreetype.so.6.20.2.p/src_svg_svg.c.o libfreetype.so.6.20.2.p/src_sdf_sdf.c.o libfreetype.so.6.20.2.p/src_cache_ftcache.c.o libfreetype.so.6.20.2.p/src_gzip_ftgzip.c.o libfreetype.so.6.20.2.p/src_lzw_ftlzw.c.o libfreetype.so.6.20.2.p/src_psaux_psaux.c.o libfreetype.so.6.20.2.p/src_psnames_psnames.c.o libfreetype.so.6.20.2.p/src_base_ftbbox.c.o libfreetype.so.6.20.2.p/src_base_ftbdf.c.o libfreetype.so.6.20.2.p/src_base_ftbitmap.c.o libfreetype.so.6.20.2.p/src_base_ftcid.c.o libfreetype.so.6.20.2.p/src_base_ftfstype.c.o libfreetype.so.6.20.2.p/src_base_ftgasp.c.o libfreetype.so.6.20.2.p/src_base_ftglyph.c.o libfreetype.so.6.20.2.p/src_base_ftgxval.c.o libfreetype.so.6.20.2.p/src_base_ftmm.c.o libfreetype.so.6.20.2.p/src_base_ftotval.c.o libfreetype.so.6.20.2.p/src_base_ftpatent.c.o libfreetype.so.6.20.2.p/src_base_ftpfr.c.o libfreetype.so.6.20.2.p/src_base_ftstroke.c.o libfreetype.so.6.20.2.p/src_base_ftsynth.c.o libfreetype.so.6.20.2.p/src_base_fttype1.c.o libfreetype.so.6.20.2.p/src_base_ftwinfnt.c.o libfreetype.so.6.20.2.p/builds_unix_ftsystem.c.o libfreetype.so.6.20.2.p/src_base_ftdebug.c.o libfreetype.so.6.20.2.p/src_bzip2_ftbzip2.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libfreetype.so.6 /home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a /home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a /home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a -lm /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlidec.a /home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a -Wl,--end-group
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(inflate.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(inftrees.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(inffast.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(zutil.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(adler32.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(crc32.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a(bzlib.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a(compress.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a(crctable.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a(decompress.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a(huffman.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a(randtable.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/bzip27630913063559/p/lib/libbz2.a(blocksort.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(png.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngerror.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngget.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngmem.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngread.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngrio.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngrtran.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngrutil.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngset.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngtrans.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngwrite.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngwtran.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngwutil.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/libpnd711c4cbd3c2e/p/lib/libpng.a(pngwio.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlidec.a(decode.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlidec.a(huffman.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlidec.a(state.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlidec.a(bit_reader.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a(constants.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a(context.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a(platform.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a(shared_dictionary.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a(transform.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/brotl7a1ee101c276c/p/lib/libbrotlicommon.a(dictionary.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(deflate.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(trees.c.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: /home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(adler32.c.o): in function `adler32_combine':
adler32.c:(.text+0x657): undefined reference to `__moddi3'
/usr/bin/ld: /home/amachado/.conan2/p/b/zlibba23a9fdcfbea/p/lib/libz.a(adler32.c.o): in function `adler32_combine64':
adler32.c:(.text+0x71a): undefined reference to `__moddi3'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

freetype/2.13.3: ERROR: 
Package '5382c7fd93cc143dabc99020ee69af8fd104786c' build failed
freetype/2.13.3: WARN: Build folder /home/amachado/.conan2/p/b/freet22b6a509fd61e/b/build-release
ERROR: freetype/2.13.3: Error in build() method, line 113
        meson.build()
        ConanException: Error 1 while executing
```

</details>

--- a/conan/tools/meson/toolchain.py
+++ b/conan/tools/meson/toolchain.py
@@ -9,7 +9,7 @@ from conan.internal.internal_tools import raise_on_universal_arch
 from conan.tools.apple.apple import is_apple_os, apple_min_version_flag, \
     resolve_apple_flags
 from conan.tools.build.cross_building import cross_building
-from conan.tools.build.flags import libcxx_flags
+from conan.tools.build.flags import libcxx_flags, architecture_flag
 from conan.tools.env import VirtualBuildEnv
 from conan.tools.meson.helpers import *
 from conan.tools.microsoft import VCVars, msvc_runtime_flag
@@ -210,6 +210,8 @@ class MesonToolchain:
         #: List of extra preprocessor definitions. Added to ``c_args`` and ``cpp_args`` with the
         #: format ``-D[FLAG_N]``.
         self.extra_defines = []
+        #: Architecture flag deduced by Conan and added to ``c_args``, ``cpp_args``, ``c_link_args`` and ``cpp_link_args``
+        self.arch_flag = architecture_flag(self._conanfile)  # https://github.com/conan-io/conan/issues/17624
         #: Dict-like object that defines Meson ``properties`` with ``key=value`` format
         self.properties = {}
         #: Dict-like object that defines Meson ``project options`` with ``key=value`` format
@@ -429,9 +431,9 @@ class MesonToolchain:
         sys_root = [f"--sysroot={self._sys_root}"] if self._sys_root else [""]
         ld = sharedlinkflags + exelinkflags + linker_script_flags + sys_root + self.extra_ldflags
         return {
-            "cxxflags": cxxflags + sys_root + self.extra_cxxflags,
-            "cflags": cflags + sys_root + self.extra_cflags,
-            "ldflags": ld,
+            "cxxflags": [self.arch_flag] + cxxflags + sys_root + self.extra_cxxflags,
+            "cflags": [self.arch_flag] + cflags + sys_root + self.extra_cflags,
+            "ldflags": [self.arch_flag] + ld,
             "defines": [f"-D{d}" for d in (defines + self.extra_defines)]
         }
 

--- a/test/integration/toolchains/meson/test_mesontoolchain.py
+++ b/test/integration/toolchains/meson/test_mesontoolchain.py
@@ -98,10 +98,10 @@ def test_extra_flags_via_conf():
 
     t.run("install . -pr:h=profile -pr:b=profile")
     content = t.load(MesonToolchain.native_filename)
-    assert "cpp_args = ['-flag0', '-other=val', '-flag1', '-flag2', '-Ddefine1=0', '-D_GLIBCXX_USE_CXX11_ABI=0']" in content
-    assert "c_args = ['-flag0', '-other=val', '-flag3', '-flag4', '-Ddefine1=0']" in content
-    assert "c_link_args = ['-flag0', '-other=val', '-flag5', '-flag6']" in content
-    assert "cpp_link_args = ['-flag0', '-other=val', '-flag5', '-flag6']" in content
+    assert "cpp_args = ['-flag0', '-other=val', '-m64', '-flag1', '-flag2', '-Ddefine1=0', '-D_GLIBCXX_USE_CXX11_ABI=0']" in content
+    assert "c_args = ['-flag0', '-other=val', '-m64', '-flag3', '-flag4', '-Ddefine1=0']" in content
+    assert "c_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6']" in content
+    assert "cpp_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6']" in content
 
 
 def test_extra_flags_via_toolchain():
@@ -138,10 +138,32 @@ def test_extra_flags_via_toolchain():
             "profile": profile})
     t.run("install . -pr:h=profile -pr:b=profile")
     content = t.load(MesonToolchain.native_filename)
-    assert "cpp_args = ['-flag0', '-other=val', '-flag1', '-flag2', '-Ddefine1=0', '-D_GLIBCXX_USE_CXX11_ABI=0']" in content
-    assert "c_args = ['-flag0', '-other=val', '-flag3', '-flag4', '-Ddefine1=0']" in content
-    assert "c_link_args = ['-flag0', '-other=val', '-flag5', '-flag6']" in content
-    assert "cpp_link_args = ['-flag0', '-other=val', '-flag5', '-flag6']" in content
+    assert "cpp_args = ['-flag0', '-other=val', '-m64', '-flag1', '-flag2', '-Ddefine1=0', '-D_GLIBCXX_USE_CXX11_ABI=0']" in content
+    assert "c_args = ['-flag0', '-other=val', '-m64', '-flag3', '-flag4', '-Ddefine1=0']" in content
+    assert "c_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6']" in content
+    assert "cpp_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6']" in content
+
+
+def test_custom_arch_flag_via_toolchain():
+    t = TestClient()
+    conanfile = textwrap.dedent("""
+    from conan import ConanFile
+    from conan.tools.meson import MesonToolchain
+    class Pkg(ConanFile):
+        settings = "os", "compiler", "arch", "build_type"
+        def generate(self):
+            tc = MesonToolchain(self)
+            tc.arch_flag = "-mmy-flag"
+            tc.generate()
+    """)
+    t.save({"conanfile.py": conanfile})
+    t.run("install .")
+    content = t.load(MesonToolchain.native_filename)
+    assert re.search(r"c_args =.+-mmy-flag.+", content)
+    assert re.search(r"c_link_args =.+-mmy-flag.+", content)
+    assert re.search(r"cpp_args =.+-mmy-flag.+", content)
+    assert re.search(r"cpp_link_args =.+-mmy-flag.+", content)
+
 
 
 def test_linker_scripts_via_conf():
@@ -169,8 +191,8 @@ def test_linker_scripts_via_conf():
 
     t.run("install . -pr:b=profile -pr=profile")
     content = t.load(MesonToolchain.native_filename)
-    assert "c_link_args = ['-flag0', '-other=val', '-flag5', '-flag6', '-T\"/linker/scripts/flash.ld\"', '-T\"/linker/scripts/extra_data.ld\"']" in content
-    assert "cpp_link_args = ['-flag0', '-other=val', '-flag5', '-flag6', '-T\"/linker/scripts/flash.ld\"', '-T\"/linker/scripts/extra_data.ld\"']" in content
+    assert "c_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6', '-T\"/linker/scripts/flash.ld\"', '-T\"/linker/scripts/extra_data.ld\"']" in content
+    assert "cpp_link_args = ['-flag0', '-other=val', '-m64', '-flag5', '-flag6', '-T\"/linker/scripts/flash.ld\"', '-T\"/linker/scripts/extra_data.ld\"']" in content
 
 
 def test_correct_quotes():

